File: StudentHousing/AdvertisingService.BusinessLogic/DataTransferObjects/AdvertisementDetailsDTO.cs

using AdvertisingService.BusinessLogic.Converters;
using System.Text.Json.Serialization;

namespace AdvertisingService.BusinessLogic.DataTransferObjects
{
    public class AdvertisementDetailsDTO
    {
        public int Id { get; set; }
        public string? CategoryName { get; set; }
        public string? Region { get; set; }
        public int PostalCode { get; set; }
        public string? City { get; set; }
        public string? District { get; set; }
        public string? StreetName { get; set; }
        public string? StreetNumber { get; set; }
        public string? UnitNumber { get; set; }
        public float NumberOfRooms { get; set; }
        public float Size { get; set; }
        public bool Furnished { get; set; }
        public bool Parking { get; set; }
        public string? Description { get; set; }
        public int  MonthlyPrice { get; set; }
        //[JsonConverter(typeof(ByteArrayConverter))]
        public byte[]? Image { get; set; }
    }
}

File: ApartmentFinder/ApartmentFinder.ClientApp/src/models/advertisement/advertisementDetailsDto.ts
export interface AdvertisementDetailsDto {
  id: number;
  categoryName: string;
  region: string;
  postalCode: number;
  city: string;
  district?: string;
  streetName: string;
  streetNumber: string;
  unitNumber: string;
  numberOfRooms: number;
  size: number;
  furnished: boolean;
  parking: boolean;
  description: string;
  monthlyPrice: number;
  images: Uint8Array[];
  advertiserId: number;
}
